File: tests/fixtures/swift/Basic.swift
import Foundation
import UIKit

protocol Repository {
    func findById(id: String) -> Any?
    func save(entity: Any) -> Any
    func deleteById(id: String)
}

struct User {
    let id: String
    var name: String
    var email: String
}

class UserService {
    var repository: Repository

    init(repository: Repository) {
        self.repository = repository
    }

    func getUser(id: String) -> User? {
        return repository.findById(id: id) as? User
    }

    func createUser(name: String, email: String) -> User {
        let user = User(id: UUID().uuidString, name: name, email: email)
        _ = repository.save(entity: user)
        return user
    }

    func deleteUser(id: String) {
        repository.deleteById(id: id)
    }
}

enum UserRole {
    case admin
    case user
    case guest
}

typealias UserID = String

extension UserService {
    func getAllUsers() -> [User] {
        return []
    }
}

func main() {
    let service = UserService(repository: InMemoryRepository())
    let user = service.createUser(name: "Alice", email: "alice@example.com")
    print(user.name)
}

File: tests/fixtures/kotlin/Basic.kt
package com.example.myapp

import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.map
import java.util.UUID as JavaUUID

data class User(val id: String, val name: String, val email: String)

sealed class Result<out T> {
    data class Success<T>(val data: T) : Result<T>()
    data class Error(val message: String) : Result<Nothing>()
}

interface Repository<T> {
    fun findById(id: String): T?
    fun save(entity: T): T
    fun deleteById(id: String)
}

object AppConfig {
    val version = "1.0.0"

    fun getEnvironment(): String {
        return System.getenv("ENV") ?: "development"
    }
}

class UserService(private val repository: Repository<User>) {
    companion object {
        fun create(repository: Repository<User>): UserService {
            return UserService(repository)
        }
    }

    fun getUser(id: String): Result<User> {
        val user = repository.findById(id)
        return if (user != null) {
            Result.Success(user)
        } else {
            Result.Error("User not found")
        }
    }

    fun createUser(name: String, email: String): Result<User> {
        val user = User(
            id = JavaUUID.randomUUID().toString(),
            name = name,
            email = email
        )
        val saved = repository.save(user)
        return Result.Success(saved)
    }

    fun deleteUser(id: String) {
        repository.deleteById(id)
    }
}

enum class UserRole {
    ADMIN,
    USER,
    GUEST
}

fun main() {
    val config = AppConfig
    println(config.getEnvironment())
    println(AppConfig.version)
}
